apps/server/lib/lexical/server/project/progress/state.ex

```diff
@@ -42,13 +44,21 @@ defmodule Lexical.Server.Project.Progress.State do
   end
 
   defp write_work_done(token) do
-    progress = Notifications.WorkDone.Progress.Create.new(token: token)
-    Transport.write(progress)
+    if client_support_work_done?() do
+      progress = Requests.CreateWorkDoneProgress.new(id: Id.next_request_id(), token: token)
+      Transport.write(progress)
+    end
   end
 
   defp write(%{token: token} = progress) when not is_nil(token) do
-    progress |> Value.to_protocol() |> Transport.write()
+    if client_support_work_done?() do
+      progress |> Value.to_protocol() |> Transport.write()
+    end
   end
 
   defp write(_), do: :ok
+
+  defp client_support_work_done? do
```

Comment:
nit: `client_supports_work_done?` but I feel that this should be done inside the configuration so we can have:
```elixir
Configuration.supports?(:work_done_progress)
```
and the implementation would be:
```elixir
@supports_keys ~w(:work_done_progress)a

def supports?(key) when key in @supports_keys do 
  get_in(Configuration.get(), [:support, key], false)
end
```